### ISSUE TITLE
Keep track of changes since the original nonlib

### ISSUE THREAD

**Issue by jean-emmanuel** [Stazed/nonlib-xt#1]
It would be sane to recreate this repo based on a clone of the unsplitted non-mixer-xt repo in order to preserve the code history (who knows if some changes of mine won't cause problems in the future ? Or if some part of the code only makes sense when reading the commit that introduced it ? Having the logs at hand will be needed eventually). You could simply remove everything except what's in nonlib/, or if you really want to optimize the git repository and keep only the commits linked to these files it's possible to purge all the rest from git's history. Here's a script to generate a clean nonlib-xt repo:

```bash
# clone
git clone https://github.com/Stazed/non-mixer-xt nonlib-xt
cd nonlib-xt
# reset to before repository split
git reset --hard 7e238c9d4221e29a472161ef397d56a7f7763a97

# remove all files except nonlib/ and purge git history 
# removes 600+ commits and ~10M
# @ https://blog.ostermiller.org/removing-and-purging-files-from-git-history/
git filter-branch --tag-name-filter cat --index-filter 'git rm -r --cached --ignore-unmatch arch  CMakeLists.txt  cmake_uninstall.cmake.in  COPYING  FL  lib  mixer  OSC.md  README.md  sedcmd  sedcmd_patch  sedinvoke' --prune-empty -f -- --all
rm -rf .git/refs/original/
git reflog expire --expire=now --all
git gc --aggressive --prune=now

# move nonlib file to top level
mv nonlib tmp && mv tmp/* ./ && rm -rf tmp
git add .
git commit -m "move nonlib files to top level"
# add readme
wget https://raw.githubusercontent.com/Stazed/nonlib-xt/main/README.md
git add README.md && git commit -m "add reamde"
```

I would recommend skipping the middle block (and run git rm everything except nonlib/ instead) and keeping the extra commits / history that may help by giving context on prior changes that happened in the lib (some commit did affect both the lib and other non-* components, purging will keep them but only with the diff that concerns lib files).

**Comment by Stazed** [comment]
I really want to reduce the download size and endless bloat and your script seems to leave enough history.  I guess it would not be enough to simply add a comment to the README.md listing git commit #7e238c9d4221e29a472161ef397d56a7f7763a97 of non-mixer-xt as the starting point? Anyone who really wanted to get the history could then find it...

Anyway I will use your script if adding a comment won't suffice. Will do it before the next release which may be a while.

**Comment by jean-emmanuel** [comment]
> I really want to reduce the download size and endless bloat and your script seems to leave enough history. 

In that case the above script seems like a good compromise. With only a commit hash in the readme it will still likely remain cumbersome to dig through the lib's history and somewhat counter-intuitive to have to look in the history of a repository that does not contain the files anymore. Anyway, no pressure, I just stumbled on the repo split recently and felt I had to share my 2 cents ;).

**Comment by Stazed** [comment]
I decided that I am going to just put a note in the REDME pointing to the library origin. 
The reason I did the submodule was to reduce duplication and some bloat. So it really would cause me pain to have actually increased duplication and bloat by doing it. I would rather revert the submodule.

It is good that you brought up the issue as having no indication of where the library came from was an oversight on my part.
